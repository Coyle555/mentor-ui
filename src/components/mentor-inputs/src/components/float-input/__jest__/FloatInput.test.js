import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitForElement, cleanup, act, getByTestId } from '@testing-library/react';

import FloatInput from '../floatInput';

test('<FloatInput /> with no props', () => {

	 const component = renderer.create( <FloatInput/> );

	 const tree = component.toJSON();
	 expect(tree).toMatchSnapshot();
});

test('Float Input with custom placeholder', () => {
	const tree = renderer.create(<FloatInput placeholder="Test placeholder" />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('<FloatInput /> with a custom className', () => {
	const component = renderer.create( <FloatInput className="custom-class"/> );

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();	
});

test('<FloatInput /> accepts a valid float value', () => {
	const component = renderer.create( <FloatInput value="3.2" /> );

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();	

});

test('Float input with required prop', () => {
	const tree = renderer.create(<FloatInput required={true} />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Float input with an invalid float', () => {
	const component = renderer.create( <FloatInput value="3.2aa" /> );

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();	
});

test('Float input blur with a preicison', () => {
	const onBlur = jest.fn();
	const { debug, container } = render(<FloatInput name="foo" onBlur={onBlur} precision={2} />);

	fireEvent.change(container.querySelector('input'), { target: { value: '123.423' } });
	fireEvent.blur(container.querySelector('input'));
	expect(onBlur).toHaveBeenCalledWith(true, '123.423', 'foo');
});

test('Minimum value passed for float input', () => {
	const component = renderer.create(<FloatInput min={5} value={5} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();	
});

test('Minimum value failed for float input', () => {
	const component = renderer.create(<FloatInput min={10} value={5} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();	
});

test('Maximum value passed for float input', () => {
	const component = renderer.create(<FloatInput max={5} value={5} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();	
});

test('Maximum value failed for float input', () => {
	const component = renderer.create(<FloatInput max={0} value={5} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();	
});
