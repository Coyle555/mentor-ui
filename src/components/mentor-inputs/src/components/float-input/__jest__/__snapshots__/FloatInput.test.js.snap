// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FloatInput /> accepts a value prop of type number 1`] = `
<input
  className="apm-mi-form-control"
  onBlur={[Function]}
  onChange={[Function]}
  type="number"
  value={3.2}
/>
`;

exports[`<FloatInput /> disabled w/ a min of 1 and max of 5 1`] = `
<input
  className="apm-mi-form-control"
  disabled={true}
  max={5}
  min={1}
  onBlur={[Function]}
  onChange={[Function]}
  type="number"
  value=""
/>
`;

exports[`<FloatInput /> with a custom className 1`] = `
<input
  className="apm-mi-form-control custom-class"
  onBlur={[Function]}
  onChange={[Function]}
  type="number"
  value=""
/>
`;

exports[`<FloatInput /> with no props 1`] = `
<input
  className="apm-mi-form-control"
  onBlur={[Function]}
  onChange={[Function]}
  type="number"
  value=""
/>
`;
